src/blob_gc_job.cc

```diff
@@ -433,32 +483,83 @@ Status BlobGCJob::RewriteValidKeyToLSM() {
   WriteOptions wo;
   wo.low_pri = true;
   wo.ignore_missing_column_families = true;
-  for (auto& write_batch : rewrite_batches_) {
-    if (blob_gc_->GetColumnFamilyData()->IsDropped()) {
-      s = Status::Aborted("Column family drop");
+  // @TODO(tabokie): fix statistics here.
+  switch (gc_rewrite_mode_) {
+    case TitanGcRewriteMode::kDefault:
+      for (auto& write_batch : rewrite_batches_) {
+        if (blob_gc_->GetColumnFamilyData()->IsDropped()) {
+          s = Status::Aborted("Column family drop");
+          break;
+        }
+        if (IsShutingDown()) {
+          s = Status::ShutdownInProgress();
+          break;
+        }
+        s = db_impl->WriteWithCallback(wo, &write_batch.first,
+                                       &write_batch.second);
+        if (s.ok()) {
+          // count written bytes for new blob index.
+          metrics_.bytes_written += write_batch.first.GetDataSize();
+          metrics_.gc_num_keys_relocated++;
+          metrics_.gc_bytes_relocated += write_batch.second.blob_record_size();
+          // Key is successfully written to LSM.
+        } else if (s.IsBusy()) {
+          metrics_.gc_num_keys_overwritten++;
+          metrics_.gc_bytes_overwritten +=
+              write_batch.second.blob_record_size();
+          // The key is overwritten in the meanwhile. Drop the blob record.
+        } else {
+          // We hit an error.
+          break;
+        }
+        // count read bytes in write callback
+        metrics_.bytes_read += write_batch.second.read_bytes();
+      }
       break;
-    }
-    if (IsShutingDown()) {
-      s = Status::ShutdownInProgress();
+    case TitanGcRewriteMode::kMerge:
+      for (auto& write_batch : rewrite_batches_without_callback_) {
+        if (blob_gc_->GetColumnFamilyData()->IsDropped()) {
+          s = Status::Aborted("Column family drop");
+          break;
+        }
+        if (IsShutingDown()) {
+          s = Status::ShutdownInProgress();
+          break;
+        }
+        s = db_impl->Write(wo, &write_batch);
+        if (s.ok()) {
+          // count written bytes for new blob index.
+          metrics_.bytes_written += write_batch.GetDataSize();
+          metrics_.gc_num_keys_relocated++;
+          // metrics_.gc_bytes_relocated +=
+          // write_batch.second.blob_record_size();
+          // Key is successfully written to LSM.
+        } else if (s.IsBusy()) {
```

Comment:
seems `IsBusy` doesn't mean `overwritten in the meanwhile` for writing with merge ops.